tests/phpunit/tests/block-template.php

```diff
@@ -275,6 +275,89 @@ public function test_get_the_block_template_html_does_not_generally_enforce_loop
 		$this->assertSame( array( false, true ), $in_the_loop_logs, 'Main query loop was triggered incorrectly' );
 	}
 
+	/**
+	 * @ticket 58319
+	 *
+	 * @covers ::get_block_theme_folders
+	 *
+	 * @dataProvider data_get_block_theme_folders
+	 *
+	 * @param string   $theme    The theme's stylesheet.
+	 * @param string[] $expected The expected associative array of block theme folders.
+	 * @param string   $message The message for when an assertion fails.
+	 */
+	public function test_get_block_theme_folders( $theme, $expected, $message ) {
+		$wp_theme   = wp_get_theme( $theme );
+		$reflection = new ReflectionMethod( $wp_theme, 'cache_get' );
+		$reflection->setAccessible( true );
+		$wp_theme->cache_delete(); // Clear cache.
+
+		$this->assertSame( $expected, get_block_theme_folders( $theme ), $message );
+		$theme_cache  = $reflection->invoke( $wp_theme, 'theme' );
+		$cached_value = $theme_cache['block_template_folders'];
+		$reflection->setAccessible( false );
+
+		$this->assertSame( $expected, $cached_value, $message );
```

Comment:
Both assertions in this test have the same `$message` argument. The `$message` parameter is used to provide context to an assertion in a multi-assertion test, so having the same `$message` argument negates the benefits.

See [Writing PHPUnit Tests - Using Assertions](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#using-assertions) in the handbook.

We can prefix this:
```suggestion
		$this->assertSame( $expected, $cached_value, "The cached value is incorrect: $message" );
```

Reply:
@costdev I recommended removing the prefix, as this is confusing.

Reply:
The messages for each assertion must be different, otherwise they don't tell us which assertion failed. Given that an assertion failure's output includes the diff between the expected and actual values, I think we could just hardcode this message. This is what's usually done in the test suite anyway.

i.e.
```php
$this->assertSame( $expected, $cached_value, 'The cached value is incorrect.' );
```

Reply:
@kt-12 Let's just hard code it.